sdk/azidentity/azure_developer_cli_credential.go

```diff
@@ -0,0 +1,168 @@
+	// to TenantID. Add the wildcard value "*" to allow the credential to acquire tokens for any tenant the
+	// logged in account can access.
+	AdditionallyAllowedTenants []string
+	// TenantID identifies the tenant the credential should authenticate in.
+	// Defaults to the azd environment, which is the tenant where selected Azure subscription is.
+	TenantID string
+
+	tokenProvider azureDeveloperCLITokenProvider
+}
+
+// init returns an instance of AzureDeveloperCLICredentialOptions initialized with default values.
+func (o *AzureDeveloperCLICredentialOptions) init() {
+	if o.tokenProvider == nil {
+		o.tokenProvider = defaultAzdTokenProvider
+	}
+}
+
+// AzureDeveloperCLICredential authenticates as the identity logged in to the [Azure Developer CLI].
+//
+// [Azure Developer CLI]: https://learn.microsoft.com/azure/developer/azure-developer-cli/overview
+type AzureDeveloperCLICredential struct {
+	s             *syncer
+	tokenProvider azureDeveloperCLITokenProvider
+}
+
+// NewAzureDeveloperCLICredential constructs an AzureDeveloperCLICredential. Pass nil to accept default options.
+func NewAzureDeveloperCLICredential(options *AzureDeveloperCLICredentialOptions) (*AzureDeveloperCLICredential, error) {
+	cp := AzureDeveloperCLICredentialOptions{}
+	if options != nil {
+		cp = *options
+	}
+	cp.init()
+	c := AzureDeveloperCLICredential{tokenProvider: cp.tokenProvider}
+	c.s = newSyncer(
+		credNameAzureDeveloperCLI,
+		cp.TenantID,
+		c.requestToken,
+		nil, // this credential doesn't have a silent auth method because the CLI handles caching
+		syncerOptions{AdditionallyAllowedTenants: cp.AdditionallyAllowedTenants},
+	)
+	return &c, nil
+}
+
+// GetToken requests a token from the Azure Developer CLI. This credential doesn't cache tokens, so every call invokes the CLI.
+// This method is called automatically by Azure SDK clients.
+func (c *AzureDeveloperCLICredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	return c.s.GetToken(ctx, opts)
+}
+
+func (c *AzureDeveloperCLICredential) requestToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	b, err := c.tokenProvider(ctx, opts.Scopes, opts.TenantID)
+	if err != nil {
+		return azcore.AccessToken{}, err
+	}
+	at, err := c.createAccessToken(b)
+	if err != nil {
+		return azcore.AccessToken{}, err
+	}
+	return at, nil
+}
+
+var defaultAzdTokenProvider azureDeveloperCLITokenProvider = func(ctx context.Context, scope []string, tenantID string) ([]byte, error) {
+	// set a default timeout for this authentication iff the application hasn't done so already
+	var cancel context.CancelFunc
+	if _, hasDeadline := ctx.Deadline(); !hasDeadline {
+		ctx, cancel = context.WithTimeout(ctx, timeoutAzdRequest)
+		defer cancel()
+	}
+
+	commandLine := "azd auth token -o json"
+	if tenantID != "" {
+		commandLine += " --tenant-id " + tenantID
+	}
+
+	for _, scopeName := range scope {
+		commandLine += " --scope " + scopeName
```

Comment:
In `AzureCLICredential` we validate that the scope is well formed.  Do we need to do that here?

Reply:
Yes! Thanks for asking

Reply:
Well, we can validate it for sure, but the error we get back from the Auth-Service (entra) is also good enough for a bad scope:

![image](https://github.com/Azure/azure-sdk-for-go/assets/24213737/e3475459-058c-4376-a4fb-d509348f5578)

The error is clear about the scope format.

My understanding is that we do validate for the azCliCredential because az supports ADD 1.0 and 2.0, and there are some convertions between resouce and scope that is done by the credential (which makes sense to validate).

But azd supports only ADD 2.0